context/context.go

```diff
@@ -38,138 +32,136 @@ func ResolveRemotesToRepos(remotes Remotes, client *api.Client, base string) (Re
 		if err != nil {
 			return result, err
 		}
-		result.BaseOverride = baseOverride
+		result.baseOverride = baseOverride
 	}
 
-	foundBaseOverride := false
-	var hostname string
+	return result, nil
+}
+
+func resolveNetwork(result *ResolvedRemotes) error {
 	var repos []ghrepo.Interface
-	for i, r := range remotes {
-		if i == 0 {
-			hostname = r.RepoHost()
-		} else if !strings.EqualFold(r.RepoHost(), hostname) {
-			// ignore all remotes for a hostname different to that of the 1st remote
-			continue
-		}
+	for _, r := range result.remotes {
 		repos = append(repos, r)
-		if baseOverride != nil && ghrepo.IsSame(r, baseOverride) {
-			foundBaseOverride = true
-		}
 		if len(repos) == maxRemotesForLookup {
 			break
 		}
 	}
-	if baseOverride != nil && !foundBaseOverride {
-		// additionally, look up the explicitly specified base repo if it's not
-		// already covered by git remotes
-		repos = append(repos, baseOverride)
-	}
 
-	networkResult, err := api.RepoNetwork(client, repos)
-	if err != nil {
-		return result, err
-	}
-	result.Network = networkResult
-	return result, nil
+	networkResult, err := api.RepoNetwork(result.apiClient, repos)
+	result.network = &networkResult
+	return err
 }
 
 type ResolvedRemotes struct {
-	BaseOverride ghrepo.Interface
-	Remotes      Remotes
-	Network      api.RepoNetworkResult
+	baseOverride ghrepo.Interface
+	remotes      Remotes
+	network      *api.RepoNetworkResult
 	apiClient    *api.Client
 }
 
-// BaseRepo is the first found repository in the "upstream", "github", "origin"
-// git remote order, resolved to the parent repo if the git remote points to a fork
-func (r ResolvedRemotes) BaseRepo() (*api.Repository, error) {
-	if r.BaseOverride != nil {
-		for _, repo := range r.Network.Repositories {
-			if repo != nil && ghrepo.IsSame(repo, r.BaseOverride) {
-				return repo, nil
+func (r *ResolvedRemotes) BaseRepo(io *iostreams.IOStreams) (ghrepo.Interface, error) {
+	if r.baseOverride != nil {
+		return r.baseOverride, nil
+	}
+
+	// if any of the remotes already has a resolution, respect that
```

Comment:
Ideally, it seems that there should only be one resolved remote for any repo, but do we need any safeguards/messaging in place for the case when there are more than one?

Reply:
Possibly we should sort the remotes list so that if there are multiple resolved remotes we will deterministically select the same one every time.

Reply:
@samcoe Good catch! The remote list is already guaranteed to be sorted: first `upstream`, then `github`, then `origin`, then alphabetical order. This makes me feel confident that the resolution will be deterministic.

Good question about what if multiple remotes have the resolution bit. Right now I would say that this is not possible, since the resolution bit is only written when no git remote already has a bit.